Assignment4/JingleiXu97-22A322/task07.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/github/JingleiXu97/Curso2022-2023/blob/master/Assignment4/course_materials/notebooks/Task07.ipynb
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+NS = Namespace("http://somewhere#")
+
+
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    ?Subject rdfs:subClassOf ns:Person. 
+  }
+  ''',
+  initNs = { "rdfs": RDFS, "ns": NS}
+)
+
+for r in g.query(q1):
+  print(r.Subject)
+
+for s,p,o in g.triples((None, RDFS.subClassOf, NS.Person)):
+  print(s)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+NS = Namespace("http://somewhere#")
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+q2 = prepareQuery('''
+  SELECT ?Subject WHERE { 
+    {?Subject rdf:type ns:Person.} UNION
+    {?x rdfs:subClassOf ns:Person.
```

Comment:
use * to retrieve all subclasses from Person (including subclasses of subclasses)